### PR TITLE
nfs for mac

### PR DESCRIPTION
**Pull request by devalih**
To improve the performance issues found with both `dev.sync.up` and `dev.up`.

NFS is the best method we've found so far for the heavy Open edX devstack on MacOSX.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,22 +72,36 @@ dev.editable-envs:  ## Copy env files outside the docker containers so it's edit
...
 
 dev.up: | check-memory ## Bring up all services with host volumes
-	docker-compose -f docker-compose.yml -f docker-compose-host.yml -f docker-compose-themes.yml -f docker-compose-watchers.yml up -d
+	docker-compose -f docker-compose.yml -f docker-compose-host.yml up -d
+
+dev.nfs.setup:  ## set's up an nfs server on the /Users folder, allowing nfs mounting on docker
+	./setup_native_nfs_docker_osx.sh
+
+dev.nfs.up: | check-memory ## Bring up all services with host volumes
+	docker-compose -f docker-composedraak_dev_routere.yml -f docker-compose-host-nfs.yml up -d
```
Comment (OmarEdraak) [inline Makefile]:
```suggestion
	docker-compose -f docker-compose.yml -f docker-compose-host-nfs.yml up -d
```


I couldn't find where `composedraak_dev_routere` is.

File: Makefile [hunk 2]
```diff
@@ -69,22 +72,36 @@ dev.editable-envs:  ## Copy env files outside the docker containers so it's edit
...
+
+dev.provision.nfs.run: ## Provision all services with local mounted directories
+	DOCKER_COMPOSE_FILES="-f docker-compose.yml -f docker-compose-host-nfs.yml" ./provision.sh
+
 dev.up.watchers: | check-memory ## Bring up asset watcher containers
 	docker-compose -f docker-compose-watchers.yml up -d
 
 dev.up.xqueue: | check-memory ## Bring up xqueue, assumes you already have lms running
-	docker-compose -f docker-compose.yml -f docker-compose-xqueue.yml -f docker-compose-host.yml up -d
+	docker-compose -f docker-compose.yml -f  docker-compose-xqueue.yml -f docker-compose-host.yml up -d
```
Comment (OmarEdraak) [inline Makefile]:
```suggestion
	docker-compose -f docker-compose.yml -f docker-compose-xqueue.yml -f docker-compose-host.yml up -d
```

File: Makefile [hunk 3]
```diff
@@ -69,22 +72,36 @@ dev.editable-envs:  ## Copy env files outside the docker containers so it's edit
...
 
 dev.up.all: | dev.up dev.up.watchers ## Bring up all services with host volumes, including watchers
 
 dev.sync.daemon.start: ## Start the docker-sycn daemon
 	docker-sync start
 
-dev.sync.provision: | dev.sync.daemon.start dev.provision ## Provision with docker-sync enabled
+dev.sync.provision: | dev.sync.daemon.start check-memory dev.clone dev.provision.sync.run stop ## Provision dev environment with all services stopped
```
Comment (OmarEdraak) [inline Makefile]:
```suggestion
dev.sync.provision: | dev.sync.daemon.start dev.provision ## Provision with docker-sync enabled
```

Right?

File: README.rst [hunk 4]
```diff
@@ -375,7 +391,7 @@ BUILD COMMANDS:
 
     git checkout master
     git pull
-    docker build -f docker/build/edxapp/Dockerfile . -t eu.gcr.io/openedx-231314/edraak/edxapp:hawthorn.edraak.v2
+    docker build -f docker/build/edxapp/Dockerfile . -t eu.gcr.io/openedx-231314/edraak/edxapp:hawthorn.edraak.v3
```
Comment (OmarEdraak) [inline README.rst]: Wohoow! Finally.

File: docker-compose-host-nfs.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
+version: "2.1"
+
+services:
+#  credentials:
```
Comment (OmarEdraak) [inline docker-compose-host-nfs.yml]: I suppose you'd like to keep the commented out code, right?

File: docker-compose.yml [hunk 6]
```diff
@@ -188,7 +188,7 @@ services:
       BOK_CHOY_CMS_PORT: 18131
       EDXAPP_TEST_MONGO_HOST: edx.devstack.mongo
       NO_PYTHON_UNINSTALL: 1
-    image: eu.gcr.io/openedx-231314/edraak/edxapp:hawthorn.edraak.v2
+    image: eu.gcr.io/openedx-231314/edraak/edxapp:${OPENEDX_RELEASE:-latest}
```
Comment (OmarEdraak) [inline docker-compose.yml]:
```suggestion
    image: eu.gcr.io/openedx-231314/edraak/edxapp:hawthorn.edraak.v3
```

For consistency.

File: docker-compose-host-nfs.yml (hunk 5 above)
Comment (devalih) [reply docker-compose-host-nfs.yml]: Yes

**Pull request merged**